hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -46,9 +110,40 @@ public List<String> getTransformersNames() {
   @Override
   public Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties) {
     Dataset<Row> dataset = rowDataset;
+    Schema incomingSchema = enableSchemaValidation ? sourceSchemaOpt.get() : null;
     for (Transformer t : transformers) {
-      dataset = t.apply(jsc, sparkSession, dataset, properties);
+      String suffix = transformerToPropKeySuffix.get(t);
+      TypedProperties transformerProps = properties;
+      if (StringUtils.nonEmpty(suffix)) {
+        transformerProps = new TypedProperties(properties);
+        Map<String, Object> overrideKeysMap = new HashMap<>();
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+          String key = (String) entry.getKey();
+          if (key.endsWith("." + suffix)) {
+            overrideKeysMap.put(key.substring(0, key.length() - (suffix.length() + 1)), entry.getValue());
+          }
+        }
+        transformerProps.putAll(overrideKeysMap);
+      }
+      dataset = t.apply(jsc, sparkSession, dataset, transformerProps);
+      if (enableSchemaValidation) {
+        incomingSchema = validateAndGetTransformedSchema(t, dataset, incomingSchema, jsc, sparkSession, properties);
```

Comment:
per current impl, we will error out here, even if the `schemaTransform()` method is not implemented?

Reply:
Yes, we will error out here if `org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer.Config#enableTransformerSchemaValidation` is enabled. This config was specifically added for schema validation of transformer.